turbinia/config/task_base_config.yaml

```diff
@@ -0,0 +1,27 @@
+Log2timelineTask:
+  required_flags: []
+  required_params: ['partitions', 'volumes', 'vss_stores']
+  allowed_flags: ['h', 'd', 'data', 'f', 'info', 'no_dependencies_check', 
+                  'no_vss', 'q', 't', 'troubles', 'use_markdown', 'V', 
+                  'vss_only', 'z']
+  allowed_params: ['artifact_definitions', 'artifact_filters', 
+      'artifact_filters_file', 'buffer_size', 'credential', 
+      'custom_artifact_definitions', 'hasher_file_size_limit', 
+      'hashers', 'logfile', 'parsers', 'partitions', 'preferred_year', 
+      'process_archives', 'process_memory_limit', 'profilers', 
+      'profiling_directory', 'profiling_sample_rate', 'queue_size', 
+      'sigsegv_handler', 'single_process', 'skip_compressed_streams', 
+      'status_view', 'storage_format', 'task_storage_format', 
+      'temporary_directory', 'volumes', 'vss_stores', 'worker_memory_limit', 'workers', 'yara_rules']
```

Comment:
So, it looks like these are mostly flags passed to log2timeline.py itself, but for the binary specific flags, I think we can probably simplify things by just defining one parameter that is something like `log2timeline_extra_args` so that we don't need to redefine all of these here. 

I'm less concerned about the extra flags passed to the binaries, but where I think it is important to still define the exact parameter interface for the Tasks is the variables that get used elsewhere in the actual Task code.  For example if the Task code had something like the following (completely made up):

```
if recipe['full_search']:
    # Something expensive that we don't always want to run
    self.process_extra_directories()
```